### DISCUSSION TITLE
How to use with detox

### DISCUSSION THREAD

**Discussion by pke** [react-native-async-storage/async-storage#477] [Q&A]
## Usage in a detox e2e scenario

My app has a welcome screen that is only shown once, so I save a value using `async-storage` after the first start. In my e2e tests I have scenarios that I want to simulate first starts and non-first starts.
For that I would have to modify the saved data (either delete the current value or set it to a specific value).

I see that you are also using detox in the sample project. I just don't get how this custom delegate thing works and why its only invoked on ios devices and not on Android devices.

My current app has the ability to follow deep links and I already have a deep link to clear the apps data via `myapp:reset-data`. That will internally be catched by the apps deeplink code and perform a `async-storage` remove all keys operation.
Is that basically the same what you are doing with the delegate or is there more to it?

**Comment by krizzu** [comment]
Custom delegate is there to allow native side access stored data. For Detox, what about [using launchApp api to restart the application?](https://github.com/wix/Detox/blob/master/docs/APIRef.DeviceObjectAPI.md#devicelaunchappparams) You can provide different launch options, so you can call your deeplinks or simply simulate app relaunch.

**Comment by pke** [comment]
Thanks for your reply! I know about the detox launch args but they don't solve my problem. The app saves a value using AsyncStorage and reads it at app start so it can decide whether its the first app start after install or not. Depending on that the start screen looks different.
For each e2e I do not want to go through the start screen as if it was the first app start but instead set the AsyncStorage to a specific state from within my detox scenarios (in a before step).

Maybe my app should expose a "app:AsyncStorage/key/value" deeplink to set a specific async stored value?

**Reply by Janaka-Steph** [reply]
Coming a bit late but did you find a solution?
